### DISCUSSION TITLE
Why isn't there a setting() helper?

### DISCUSSION THREAD

**Discussion by Rattone** [spatie/laravel-settings#254] [Q&A]
Is there a reason why this package doesn't include an helper to get settings for example in blade views?
A common example: I have a `footer.copyright` setting for the text in footer.

**Comment by rubenvanassche** [comment]
The whole point of the package is strongly typed settings, you would lose those with such a method, refactoring wouldn't work anymore, no IDE hints etc

Writing:

```php
app(FooterSettings::class)->copyright
```

Isn't that horrible in my opinion tbh.

**Comment by serious-angel** [comment]
It should be possible to create a custom helper function and/or Facade, of course.

For example, there's a package [Laravel Settings](<https://randallwilk.dev/docs/laravel-settings/v3/introduction>), started by Randall Wilk, that provides a helper function:

```php
if (! function_exists('settings')) {
    function settings($key = null, $default = null, $context = null)
    {
```
Source: [src/helpers.php](<https://github.com/rawilk/laravel-settings/blob/685136b23a6a26321c621a9d315f5fd7e21f5174/src/helpers.php#L8-L10>)

An example is provided in their documentation:
```php
// Store
Settings::set('foo', 'bar');
settings()->set('foo', 'bar');
settings(['foo' => 'bar']);

// Retrieving
Settings::get('foo'); // 'bar
settings()->get('foo');
settings('foo');
```
[Source](<https://randallwilk.dev/docs/laravel-settings/v3/introduction#user-content-alternatives>)

It's worth to note that they also mention Spatie's package on the same documentation page.
